[DRAFT] DOCS: Framer Motion

PedroAmorimG opened the pull request:
- [x] Geral

- [x] Components e Hooks

- [x] Motion Values

artistrea commented on Diretoria Projetos/prisma_orm/Instalacao.md:
```diff
@@ -0,0 +1,143 @@
+      POSTGRES_USER: teste
+      POSTGRES_PASSWORD: teste
+    ports:
+      - "5432:5432"
```
É incômodo mapear pra 5432 se você já tem postgres na sua máquina. Tem que resolver conflito.
```suggestion
      - "5436:5432"
```

artistrea commented on Diretoria Projetos/prisma_orm/Instalacao.md:
````diff
@@ -0,0 +1,143 @@
+
+### Definir localização do seu banco de dados
+ 
+- PATH = diretório raiz = teste
+- POSTGRES_USER = usuario definido no `docker-compose.yml` = teste
+- POSTGRES_PASSWORD = senha definida no `docker-compose.yml` = teste
+
+```bash
+#.env
+
+DATABASE_URL="postgresql://{POSTGRES_USER}:{POSTGRES_PASSWORD}@localhost:5432/{PATH}?schema=public"
+```
````
Esse PATH aí é o nome do banco de dados msm
```suggestion
- POSTGRES_DB = nome do bd definido no `docker-compose.yml` = teste
- POSTGRES_USER = usuario definido no `docker-compose.yml` = teste
- POSTGRES_PASSWORD = senha definida no `docker-compose.yml` = teste

```bash
#.env

DATABASE_URL="postgresql://{POSTGRES_USER}:{POSTGRES_PASSWORD}@localhost:5432/{POSTGRES_DB}?schema=public"
```
```

artistrea commented on Diretoria Projetos/prisma_orm/como_usar.md:
É interessante criar scripts para fazer as operações comuns (push e generate) do prisma no package.json: https://github.com/StructCE/app-interno/blob/main/packages/db/package.json

Além disso, o node não roda as coisas considerando o arquivo `.env`, precisa rodar usando o pacote `dotenv-cli` e iniciando o comando com `with-env`, que nem no exemplo do link

artistrea commented on Diretoria Projetos/prisma_orm/explicacao.md:
Adiciona um [alerta](https://retype.com/components/alert/) só pra citar que o prisma tem vários componentes. Um que não é obrigatório mas é muito útil é o prisma studio.

artistrea reviewed (changes requested):
Além das pequenas alterações que pedi, 2 coisas:
- [x] Use outros ícones além do diamond;
- [x] Coloca essa documentação em `Execução/prisma_orm`. Colocaremos coisas que não são diretamente relacionados à diretoria lá.

artistrea commented on Diretoria Projetos/prisma_orm/Instalacao.md:
```diff
@@ -0,0 +1,143 @@
+---
+
+<!-- Araújo -->
+
```
```suggestion
É importante dizer que a instalação do prisma deve ser feito no projeto, e não na própria máquina.
```

EdPPF committed:
Tudo iniciado e finalizado

PedroAmorimG committed:
Correção, Server Actions, caching e optimizações

PedroAmorimG committed:
Correções feitas.

PedroAmorimG committed:
Correção otimização

PedroAmorimG committed:
Divisão em páginas, mudanças nos icones, nomes e ordem

marqueswill committed:
Changes to be committed:
	new file:   retype.yml
	new file:   stack/next-auth/callbacks.md
	renamed:    stack/next-auth/explicacao.md -> stack/next-auth/first-steps.md
	deleted:    stack/next-auth/implementacao.md
	new file:   stack/next-auth/providers.md
	new file:   stack/next-auth/routes.md
	new file:   stack/next-auth/session.md

EdPPF committed:
Adicionando `h1` e arrumando imagens

EdPPF committed:
Adicionando infos, corrigindo typos e mudando uma estrutura

EdPPF committed:
Adicionando seções de customização de variáveis

EdPPF committed:
Adicionando seções de utilização do tailwind e várias imagens

marqueswill committed:
Changes to be committed:
	deleted:    stack/next-auth/index.yml

marqueswill committed:
Changes to be committed:
	new file:   stack/next-auth/index.yml
	deleted:    stack/retype.yml

marqueswill committed:
Changes to be committed:
	deleted:    retype.yml
	new file:   stack/retype.yml

MatheusNevs committed:
zod docs

MatheusNevs committed:
hotfix

PedroAmorimG committed:
Inicio FramerMotion

EdPPF committed:
Aplicando review - adição de imagens e correções solicitadas

PedroAmorimG committed:
Finalização da visão geral

PedroAmorimG committed:
Mudança no nome

MatheusNevs committed:
changing text

Co-authored-by: EdPPF <90335387+EdPPF@users.noreply.github.com>

MatheusNevs committed:
att

Co-authored-by: EdPPF <90335387+EdPPF@users.noreply.github.com>

MatheusNevs committed:
att

Co-authored-by: EdPPF <90335387+EdPPF@users.noreply.github.com>

MatheusNevs committed:
att

Co-authored-by: EdPPF <90335387+EdPPF@users.noreply.github.com>

pafev committed:
ajustes finais

pafev committed:
add ./retype.yml to gitignore

pafev committed:
doc intro and t3-stack

PedroAmorimG committed:
Correção

PedroAmorimG committed:
Tópico de responsividade adicionado

pafev committed:
mudando h3 da intro stack

MatheusNevs commented:
Sugestões:
- Explicar um pouco melhor os layouts, que serve para fazer com que a mudança de alguma característica
de estilização seja feita de forma animada.
- Colocar o que tá em propriedades + (Gestos, Layout, Scroll) em uma parte de animações, deixar 
keyframes por ultimo dos tópicos.
- Citar e apontar documentação para outros componentes, só dar uma breve explicadinha e redirecionar para a documentação deles.

MatheusNevs commented on stack/framer-motion/VisaoGeral.md:
```diff
@@ -0,0 +1,571 @@
+
+## Componente Motion
+
+O componente motion é a forma mais básica de criar uma animação. Para utilizá-lo, será necessário colocar `<motion.elemento>`. As animações podem ocorrem quando o componente é montado e quando ele é "destruído".
```
```suggestion
O componente motion é a forma principal de criar uma animação. Para utilizá-lo, será necessário colocar `<motion.elemento>`. As animações podem ocorrem quando o componente é montado e quando ele é "destruído".
```

MatheusNevs commented on stack/framer-motion/VisaoGeral.md:
```diff
@@ -0,0 +1,571 @@
+
+### Animate
+
+A maior parte das animações poderá ser feita utilizando o component `motion` e a prop `animation`. Agora com a `tag` criada, poderá ser atribuída a prop `animation`. Ela poderá conter propriedades do `CSS` como `x`, `y` e `rotate`que ocasionarão animações se algum desses valores for alterado.
```
```suggestion
A maior parte das animações poderá ser feita utilizando o component `motion` e a prop `animate`. Com a `tag` criada, poderá ser atribuída a prop `animate`. Ela poderá conter propriedades do `CSS` como `x`, `y` e `rotate` que as tornarão animadas a cada alteração em seus valores.
```

MatheusNevs commented on stack/framer-motion/VisaoGeral.md:
```diff
@@ -0,0 +1,571 @@
+
+!!!
+
+A animação acima ocorrerá quando o componente for montado.
```
Tornará essas propriedades animadas e, caso seus valores sejam diferentes dos na prop Initial, então haverá uma animação de entrada, isso também se o Initial não estiver marcado como false.

MatheusNevs replied:
Acho que pode explicar isso na parte que fala do initial={false}

MatheusNevs commented on stack/framer-motion/VisaoGeral.md:
```diff
@@ -0,0 +1,571 @@
+
+### Transition
+
+É possível criar variantes de uma animação utilizando a prop `transition`. Ela torna possível alterações como duração da animação, atraso e o tipo, por exemplo.
```
Citar que há transições padrões para o que é passado no animate, mas pode customizar elas pelo prop Transition.

MatheusNevs commented on stack/framer-motion/VisaoGeral.md:
```diff
@@ -0,0 +1,571 @@
+
+### Dynamic variants
+
+É possível a utilização de "variáveis" em uma `variant`. Para isso, é necessário a prop `custom` colocando o valor da variável. E para utilizá-la na `variant` deverá ser criada uma `arrow function`.
```
```suggestion
É possível a utilização de "variáveis" em uma `variant`. Para isso, é necessário a prop `custom` colocando o valor da variável. E para utilizá-la na `variant` poderá ser criada uma `arrow function`.
```

MatheusNevs commented on stack/framer-motion/VisaoGeral.md:
```diff
@@ -0,0 +1,571 @@
+
+## Resposividade
+
+Resposividade é esperada de toda UI moderna e não é diferente com as animações utilizadas com o Framer Motion. Essa característica pode ser garantida utilizando `variáveis CSS` e `media query`. Projetos com `Tailwind` disponível tornam o processo de `media query` mais fácil, contudo é importante lembrar que o `Tailwind` é focado em `mobile first`, por isso, o valor "padrão" das variáveis deverão  ...
```
```suggestion
Resposividade é esperada de toda UI moderna e não é diferente com as animações utilizadas com o Framer Motion. Essa característica pode ser garantida utilizando `variáveis CSS` e `media query`. Projetos com `Tailwind` disponível tornam o processo de `media query` mais fácil, contudo é importante lembrar que o `Tailwind` é focado em `mobile first`, por isso, o valor "padrão" das variáveis deverão ser para telas pequenas e a `media query` deverá ajustá-las. Se `Tailwind` não estiver disponível, o processo de criação e `media query` poderá ser feito no próprio CSS e a utilização no componente `motion` será feito por meio do "var(--nome-da-variavel)".
```

PedroAmorimG committed:
alterações pedidas

pafev committed:
fix pages to app  router recommendation (iniciando.md)

PedroAmorimG committed:
mods sugeridas e useAnimate

PedroAmorimG committed:
Merge branch 'main' into feat-FramerMotion

PedroAmorimG committed:
Extras

PedroAmorimG committed:
Mudança na ordeneção da página

MatheusNevs commented on stack/framer-motion/TopAvanc.md:
```diff
@@ -0,0 +1,155 @@
+### useTransform
+
+O useTransform é um dos meios que utilizam o `MotionValue` para oferecer melhor controle da animação gerada. Ele recebe um valor , range de entrada e range
+de saída e baseado nesses valores ele retorna um valor da range de saída.
```
```suggestion
O useTransform é um dos meios que utilizam o `MotionValue` para oferecer melhor controle da animação gerada. Ele recebe um `MotionValue` , range de entrada e range de saída, ou uma array de valores de acordo com sua entrada, e baseado nesses valores ele retorna um valor de saída.
```

MatheusNevs commented on stack/framer-motion/TopAvanc.md:
```diff
@@ -0,0 +1,155 @@
+de saída e baseado nesses valores ele retorna um valor da range de saída.
+
+!!!
+Se a saída mudar de comportamento a partir de certo valor da entrada poderá ser utilizado mais de 2 valores na entrada e na saída para melhor descrever esse comportamento.
```
Acho melhor falar aqui que pode ser passado uma função que recebe quantas entradas quiser e então retorna a saída de acordo com a lógica desenvolvida.

MatheusNevs commented on stack/framer-motion/TopAvanc.md:
```diff
@@ -0,0 +1,155 @@
+### Motion Values
+
+- (useVelocity)[https://www.framer.com/motion/use-velocity/]: Permite utilizar a velocidade ou acelereção de um componente para manipular alguma característica da animação.
+- (useSpring)[https://www.framer.com/motion/use-spring/]: Pode ser utilizado para criar movimentações do tipo mola.
```
```suggestion
- [LazyMotion](https://www.framer.com/motion/lazy-motion/): Permite diminuir o pacote inicial do Framer Motion e carregar "partes" do pacote quando for utilizar na página.
- [Reorder](https://www.framer.com/motion/reorder/): Componente útil para criar sequências de elementos animadas que podem ser ordenadas pelo usuário.
  
### Motion Values

- [useVelocity](https://www.framer.com/motion/use-velocity/): Permite utilizar a velocidade ou acelereção de um componente para manipular alguma característica da animação.
- [useSpring](https://www.framer.com/motion/use-spring/): Pode ser utilizado para criar movimentações do tipo mola.
```

MatheusNevs commented on stack/framer-motion/TopAvanc.md:
```diff
@@ -0,0 +1,155 @@
+
+===
+
+## useAnimate
```
Dá uma explicada nas vantagens de usar o useAnimate ao invés de usar somente uma motion.div

MatheusNevs commented:
A documentação ta nice, mas acho que às vezes pode gerar uma confusão de que parece que alguns componentes fazem a mesma coisa. Daí acho que poderia colocar uns assets, gifs, para dar uma ideia visual pra gelera. Se não puder ta tranqs, mas ficaria bem massa com os assets.

PedroAmorimG committed:
Atualização dos TopAvanc

MatheusNevs commented on stack/framer-motion/TopAvanc.md:
```diff
@@ -0,0 +1,156 @@
+
+!!!
+O useTransform pode receber vários valores para definir o range de entrada e o range de saída. De acordo com os valores da entrada e a ordem dos valores de saída, a função estabelecerá uma lógica que definirá o resultado final.
+!!!
```
```suggestion
O useTransform é um dos meios que utilizam o `MotionValue` para oferecer melhor controle da animação gerada. Ele recebe uma array com possíveis entradas, definindo o intervalo de valores possíveis para os `MotionValues`, e outras arrays, do mesmo tamanho das possibilidades de entradas, que irão conter as saídas para cada entrada.
```

MatheusNevs replied:
Acho que assim ficaria explicado melhor, tu acha?

PedroAmorimG committed:
mudança useTransform

MatheusNevs committed:
changing order

pafev committed:
Merge branch 'main' of github.com:StructCE/DocStruct into feat-FramerMotion

pafev committed:
update with main

pafev committed:
update with main
